samples/sensor/scd4x/src/main.c

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2022 Stephen Oliver
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/zephyr.h>
+#include <zephyr/logging/log.h>
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/sensor.h>
+#include <stdio.h>
+
+#include <zephyr/drivers/sensor/scd4x.h>
+
+LOG_MODULE_REGISTER(main);
+
+#if !DT_HAS_COMPAT_STATUS_OKAY(sensirion_scd4x)
+#error "No sensirion,scd4x compatible node found in the device tree"
+#endif
+
+
+void main(void)
+{
+	printk("\n");
+
+	const struct device *scd = DEVICE_DT_GET_ANY(sensirion_scd4x);
+
+	struct sensor_value temp, hum, co2;
+
+	if (!device_is_ready(scd)) {
+		LOG_ERR("Device %s is not ready", scd->name);
+		return;
+	}
+
+	while (true) {
+		if (sensor_sample_fetch(scd)) {
+			LOG_ERR("Failed to fetch sample from SCD4X device");
+			return;
+		}
+
+		sensor_channel_get(scd, SENSOR_CHAN_AMBIENT_TEMP, &temp);
+		sensor_channel_get(scd, SENSOR_CHAN_HUMIDITY, &hum);
+		sensor_channel_get(scd, SENSOR_CHAN_CO2, &co2);
+
+		double temperature = sensor_value_to_double(&temp);
+		#ifdef CONFIG_APP_USE_FAHRENHEIT
+		temperature = ((9.0 / 5.0) * temperature) + 32;
+		#endif
```

Comment:
Just a suggestion, you can do it like this as well, I heard it improves readability, at the expense of probably compiling a little bit more:

```c
		if (IS_ENABLED(CONFIG_APP_USE_FAHRENHEIT)) {
			temperature = ((9.0 / 5.0) * temperature) + 32;
		}
```